### ISSUE TITLE
Separate CLLocationManager object and delegate from individual scenes

### ISSUE THREAD

**Issue by Aramist** [ABKK-CodePath/demo-day#8]
Currently each scene that requests the user's location does so by extending itself as a CLLocationManagerDelegate and managing its own CLLocationManager object, which requires the scene to:
1. independently request the user's location, even if another scene has already done so in the application's lifetime
2. re-request the location if the view controller is deallocated and later reallocated (as is the case when doing lots of navigation in the app)
Requesting the user's location takes up to 10 seconds sometimes, so it should ideally be done only once

**Comment by Aramist** [comment]
Large progress on this in the firebase branch, but the other changes there are too bulky to immediately merge into main

**Comment by Aramist** [comment]
Being addressed in Pull Request #20

**Comment by Aramist** [comment]
Added to main branch in commit 57a6b6366653040260d1c4921e1146a5caada0cf
